### DISCUSSION TITLE
Parsing the output of an unescaping parser?

### DISCUSSION THREAD

**Discussion by VorpalBlade** [winnow-rs/winnow#545] [Q&A]
I'm using `escaped_transform` to unescape an optionally quoted string:

```rust
fn any_string(i: &mut &str) -> PResult<CompactString> {
    trace(
        "any_string",
        alt((quoted_string, unquoted_string_with_escapes)),
    )
    .parse_next(i)
}

/// Quoted string value
fn quoted_string(i: &mut &str) -> PResult<CompactString> {
    delimited(
        '"',
        escaped_transform(take_till(1.., ['"', '\\']), '\\', escapes),
        '"',
    )
    .map(|s: CompactStringWrapper| s.0)
    .parse_next(i)
}

/// Unquoted string value
fn unquoted_string_with_escapes(i: &mut &str) -> PResult<CompactString> {
    escaped_transform(take_till(1.., [' ', '\t', '\n', '\r', '\\']), '\\', escapes)
        .map(|s: CompactStringWrapper| s.0)
        .parse_next(i)
}
```

I then want to continue parsing the result of this string, as the now unquoted string contains structured data. Is this something that is supported in winnow, or should I just do it as a second stage after parsing the initial file format?

In *some* cases I also need to base64-decode it then continue parsing the inner data. This is determined by a modifier to another value much earlier in the same overall entry, (seriously, I'm not kidding, this format is awful). This is only true for the last string on every line. The other ones cannot be base64 at least.

I can imagine this comes up when parsing languages that may contain other (e.g. JS or CSS in a `<script>` tag in HTML etc). So I would assume it is supported somehow?

**Comment by epage** [comment]
Huh, there is `and_then` but haven't double checked if using a non-`recognize`d string.  You can also not use combinators and call the parsers directly.

**Reply by VorpalBlade** [reply]
Thanks I'll have a play around with this after the holidays.
